pyccel/stdlib/numpy/numpy_f90.f90

```diff
@@ -0,0 +1,145 @@
+! --------------------------------------------------------------------------------------- !
+! This file is part of Pyccel which is released under MIT License. See the LICENSE file   !
+! or go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details. !
+! --------------------------------------------------------------------------------------- !
+
+module numpy_f90
+
+    use ISO_C_BINDING
+    
+    implicit none
+    
+    private
+
+    public :: numpy_sign
+
+    interface numpy_sign
+        module procedure i8_sign
+        module procedure i16_sign
+        module procedure i32_sign
+        module procedure i64_sign
+        module procedure float_sign
+        module procedure double_sign
+        module procedure cmplx_float_sign
+        module procedure cmplx_double_sign
+    end interface numpy_sign
+    
+    contains
+
+    ! Implementation of numpy.sign function for real numbers
+    elemental function i8_sign(x) result(y)
+
+        implicit none
+
+        INTEGER(C_INT8_T), value :: x
+        INTEGER(C_INT8_T)        :: y
+        
+        y = merge(merge(-1, 1, x .lt. 0), 0, x .ne. 0)
```

Comment:
It would be better to use type specifiers for the literals in this file:
```
y = merge(merge(-1_C_INT8_T, 1_C_INT8_T, x .lt. 0_C_INT8_T), 0_C_INT8_T, x .ne. 0_C_INT8_T)
```
As you can see this is hard to read, which is why we usually rename these values when we import:
```
y = merge(merge(-1_i8, 1_i8, x .lt. 0_i8), 0_i8, x .ne. 0_i8)
```
(It may be faster to write this file in python and get pyccel to give you the fortran version, you can then modify it to add the complex case following the example that you generate)

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/b4ef1309aa2a48ad51b70675d90333c15ab270d4